Apps/WebClient/src/Server/Services/NoteService.cs

```diff
@@ -29,72 +31,145 @@ public class NoteService : INoteService
     {
         private readonly ILogger logger;
         private readonly INoteDelegate noteDelegate;
+        private readonly IProfileDelegate profileDelegate;
+        private readonly ICryptoDelegate cryptoDelegate;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="NoteService"/> class.
         /// </summary>
         /// <param name="logger">Injected Logger Provider.</param>
         /// <param name="noteDelegate">Injected Note delegate.</param>
-        public NoteService(ILogger<NoteService> logger, INoteDelegate noteDelegate)
+        /// <param name="profileDelegate">Injected Profile delegate.</param>
+        /// <param name="cryptoDelegate">Injected Crypto delegate.</param>
+        public NoteService(ILogger<NoteService> logger, INoteDelegate noteDelegate, IProfileDelegate profileDelegate, ICryptoDelegate cryptoDelegate)
         {
             this.logger = logger;
             this.noteDelegate = noteDelegate;
+            this.profileDelegate = profileDelegate;
+            this.cryptoDelegate = cryptoDelegate;
         }
 
         /// <inheritdoc />
-        public RequestResult<Note> CreateNote(Note note)
+        public RequestResult<UserNote> CreateNote(UserNote userNote)
         {
+            UserProfile profile = this.profileDelegate.GetUserProfile(userNote.HdId).Payload;
+            string? key = profile.EncryptionKey;
+            if (key == null)
+            {
+                this.logger.LogInformation($"User does not have a key: ${userNote.HdId}");
+                throw new ApplicationException("Profile key not set");
```

Comment:
Should this be an ApplicationException or a failed result which would be processed cleanly by the UI?

Reply:
I think in this particular case the error should not be returned to the user since there is not much for the user to do. If there is no key at this point is a 'catastrophic' failure that we will have to look into it. However I am not completely sold on the type of exception. Thoughts?